rbac/management/group/relation_api_dual_write_group_handler.py

```diff
@@ -123,13 +127,29 @@ def _replicate(self):
         except Exception as e:
             raise DualWriteException(e)
 
-    def replicate_added_role(self, role: Role):
-        """Replicate added role."""
+    def extend_relations_to_remove(self, relations_to_remove: list[Relationship]):
+        """Extend relations to remove in replication."""
+        if not self.replication_enabled():
+            return
+        self.group_relations_to_remove.extend(relations_to_remove)
+
+    def add_system_roles_to_default_custom_group(self, custom_group: Group):
+        """Add system roles to default custom group."""
+        roles = Role.objects.filter(policies__group=custom_group)
+        system_roles = roles.filter(tenant=Tenant.objects.get(tenant_name="public"))
```

Comment:
Curious – do you need the additional filter on tenant?

Reply:
Intent here is to fetch only system roles which are assigned to custom default group. Let me know If you think that it is not needed.